google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/BitReversedSequenceStyleGeneratorTests.java

```diff
@@ -44,4 +51,66 @@ protected Serializable generateBaseValue(
         0b1000100100000000000000000000000000000111111111111111111111111111L,
         generator.generate(session, customer));
   }
+
+  @Test
+  public void testGenerateRetriesAbortedException() {
+    JdbcAbortedDueToConcurrentModificationException jdbcAbortedException =
+        mock(JdbcAbortedDueToConcurrentModificationException.class);
+    AbortedDueToConcurrentModificationException abortedException =
+        mock(AbortedDueToConcurrentModificationException.class);
+    when(jdbcAbortedException.getCause()).thenReturn(abortedException);
+    SharedSessionContractImplementor session = mock(SharedSessionContractImplementor.class);
+    Customer customer = new Customer();
+    AtomicInteger attempt = new AtomicInteger();
+    BitReversedSequenceStyleGenerator generator =
+        new BitReversedSequenceStyleGenerator() {
+          protected Serializable generateBaseValue(
+              SharedSessionContractImplementor session, Object entity) {
+            if (attempt.addAndGet(1) == 1) {
+              throw new GenericJDBCException("Transaction was aborted", jdbcAbortedException);
+            } else {
+              return 1L;
+            }
+          }
+        };
+    assertEquals(
+        Long.reverse(1L),
+        generator.generate(session, customer));
+    assertEquals(2, attempt.get());
+  }
+
+  @Test
+  public void testGeneratePropagatesOtherExceptions() {
+    SharedSessionContractImplementor session = mock(SharedSessionContractImplementor.class);
+    Customer customer = new Customer();
+    BitReversedSequenceStyleGenerator generator =
+        new BitReversedSequenceStyleGenerator() {
+          protected Serializable generateBaseValue(
+              SharedSessionContractImplementor session, Object entity) {
+            throw mock(SpannerException.class);
+          }
+        };
+    assertThrows(SpannerException.class, () -> generator.generate(session, customer));
+  }
+
+  @Test
+  public void testGenerateGivesUpEventually() {
```

Comment:
This test should keep retrying until reach the maximum attempts, the num of retries should also be verified.

Reply:
Done